### ISSUE TITLE
Pause snapshot pruning during send

### ISSUE THREAD

**Issue by icewind1991** [jimsalterjrs/sanoid#376]
When transferring a large dataset with syncoid, it's likely that the "oldest full snapshot" will be pruned by the time it's done transfering, leaving no snapshots in common between the source and target dataset.

It would be nice if sanoid could detect that a snapshot is currently being transferred and postpone any pruning of snapshots untill the send operation is complete.

You can of course work around the problem by disabling auto prune manually during the transfer but this is easy to overlook and has already costed me multiple days of transfering

**Comment by phreaker0** [comment]
Did you actually encounter the described issue? I assumed zfs would deny destroying a snapshot if it is still used.

**Comment by icewind1991** [comment]
Yes, I've encountered it a couple of times that it gave a "target exists but has no snapshots matching" right after it finished sending the oldest snapshot

**Comment by phreaker0** [comment]
Mhm, I can't think of a reliable way to check if a snapshot is currently used in an replication. For you use case you could simply use the dedicated sync snapshots syncoid creates by default. I guess you are using "--no-sync-snap".

**Comment by icewind1991** [comment]
I'm not, it's the oldest existing snapshot that is being send and pruned, the sync snapshot is only used as newest snapshot

**Comment by phreaker0** [comment]
During the send zfs won't allow the snapshot to be pruned as it is still in use for the send process. And only after the send is complete the snapshot can be pruned, so even if the the pruning is paused while the send is ongoing it won't help with this problem.

But you could use the --create-bookmark option which would solve your problem as the replication can still be done even if the source snapshot is removed.
